keylime/src/structures/capabilities_negotiation.rs

```diff
@@ -0,0 +1,353 @@
+use serde::{Deserialize, Serialize};
+
+// Define the structure for the AttestationRequest:
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AttestationRequest {
+    #[serde(rename(serialize = "data", deserialize = "data"))]
+    pub data: RequestData,
+}
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RequestData {
+    #[serde(rename(serialize = "type", deserialize = "type"))]
+    pub type_: String,
+    pub attributes: Attributes,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Attributes {
+    pub evidence_supported: Vec<EvidenceSupported>,
+    pub boot_time: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct EvidenceSupported {
+    pub evidence_class: String,
+    pub evidence_type: String,
+    pub agent_capabilities: AgentCapabilities,
+    pub version: Option<String>,
+}
```

Comment:
`EvidenceSupported` could probably be more generic, having just the type and version fields, and then maybe an `attributes` field that are different for each type of evidence: `tpm_quote`, `ima_log`, and `mb_log`. 

`AgentCapabilities` in this structure, as it is now, is related only to the `tpm_quote`, so probably it should not be part of all the evidence types. Maybe the idea was to have the `AgentCapabilities` be the `attributes` field I suggested above.

Reply:
My question is similar to the previous one ... why not leaving Agent/TPMCapabilities and version as optional instead of using unions?

Reply:
Initially, I implemented the solution by using Option<>. I was not able to find a solution by using enums.